### PR TITLE
Few minor improvements for Waterfox Classic

### PR DESCRIPTION
**Pull request by hawkeye116477**
That should fix some problems which I encountered when building on Windows and restore user-agent overrides for Linux and playing some movies.

### CODE REVIEW

**Comment by hawkeye116477** [comment]
@MrAlex94 That should fix YouTube playback on Windows, some problems which I encountered when building on Windows and restore user-agent overrides for Linux.

File: browser/installer/package-manifest.in [hunk 1]
```diff
@@ -146,7 +146,7 @@
 @BINPATH@/@DLL_PREFIX@mozavcodec@DLL_SUFFIX@
 #endif
 @RESPATH@/browser/blocklist.xml
-#ifdef XP_WIN
+#if defined(XP_WIN) || defined(XP_LINUX)
```
Comment (grahamperrin) [inline browser/installer/package-manifest.in:149]: Can this be enhanced to include (edge case) FreeBSD? I guess, 

`#if defined(XP_WIN) || defined(XP_LINUX) || defined(BSD)`

– that is, taking a hint from https://github.com/MrAlex94/Waterfox/blob/f277dc9e4ba9ddb5064017ded4244c8d5422a059/intl/icu/source/common/unicode/platform.h#L164

**Comment by grahamperrin** [comment]
Re: https://github.com/MrAlex94/Waterfox/commit/fbb4667b0679b6984b418b06bf4fcd9471c41421#r35573351 I'll find an example of an OS-specific add-on …

**Comment by hawkeye116477** [comment]
Ah, conflict and I'll need to do that again.

**Comment by hawkeye116477** [comment]
I updated libnestegg again, but now from good revision (tested on Linux and Windows) :smile: That's needed to fix playing of some movies like that => https://bug1480003.bmoattachments.org/attachment.cgi?id=8996618.

**Comment by hawkeye116477** [comment]
Small update:
I little changed look of message about restart needed after locale selecting.
Screenshot_20191022_140726
Added also some useful features from Cyberfox: restart button in PanelUI and Menu, copying current tab URL, copying all tab URLs and new advanced preferences.
Screenshot_20191022_140926

I also fixed broken links in about dialog and made some strings translatable.

New strings are only translated in Polish, other locales contains English strings.

**Comment by hawkeye116477** [comment]
I ported now from Cyberfox what @Peacock365 requested, so option for disabling loading images with exceptions, disabling JS with advanced settings and special additions from me => exceptions for JS and option to disable notifications permanently with exceptions.
Screenshot_20191027_170928

Screenshot_20191027_093042

File: browser/installer/package-manifest.in (hunk 1 above)
Comment (grahamperrin) [inline browser/installer/package-manifest.in:149]: As this (pull request) was not merged before distribution of Waterfox Classic `2019.10 (20191023220600)`, I posted something to Reddit for the AMO use case: 

https://www.reddit.com/comments/d79h2l/-/f5htjhn/

**Comment by Peacock365** [comment]
Fantastic work, @hawkeye116477. Have you verified whether or not the new settings also show up when _browser.preferences.useOldOrganization_ is set to _false_ in Waterfox Classic? Please verify. Also, I hope of course that the new settings will also make it to Waterfox Current at some point. Thumbs up for your work so far!

**Comment by hawkeye116477** [comment]
@Peacock365 These settings are only for old organization.

**Comment by Peacock365** [comment]
@hawkeye116477 Not gonna lie, this is a bit of a bummer. I like the new organization of the settings much better, the new organization is also similar to how the settings in Waterfox Current are organized. I do wonder though, you have replicated these settings in Waterfox Current already: https://github.com/MrAlex94/Waterfox/pull/1213 Is it really that difficult to add them to the new settings page in Waterfox Classic as well? Just curious. I don't want to press the issue, but the settings being available in one settings page and not in the other is not optimal.

Furthermore, regarding the "Notifications" setting. You may have noticed that _dom.serviceWorkers.enabled_ is set to false in Waterfox Classic (and in Waterfox Current, too). That was done for good reason, as GMail refused to work in Waterfox Classic with service workers enabled. @grahamperrin, please reference the GMail issue here if possible. Web notifications rely on service workers, though. I'd say the notifications setting should be disabled by default, i.e. it should be opt-in, and if it is checked by the user, the service workers pref should be flipped to true.

**Comment by hawkeye116477** [comment]
`but the settings being available in one settings page and not in the other is not optimal.`
@Peacock365 Old settings are default, also Locale Selector isn't on new settings. For me, old settings looks better. Maybe I'll look to it later.
Also see https://github.com/MrAlex94/Waterfox/issues/494#issuecomment-398048175.

Update:
`Web notifications rely on service workers, though. `
So how it's possible that the notification on this site `https://davidwalsh.name/demo/notifications-api.php` still work despite that workers are disabled? There are 2 prefs => `dom.webnotifications.serviceworker.enabled` and `dom.webnotifications.enabled`.  Second pref enables all notifications, first only those relying on service workers, right?

Ported new things from Cyberfox:
Screenshot_20191031_000503

Screenshot_20191030_214524

**Comment by hawkeye116477** [comment]
`Web notifications rely on service workers, though. `
As I know there are 2 types of notifications => desktop notifications and service worker notifcations.
For example if you go here `https://davidwalsh.name/demo/notifications-api.php`, the notification still work despite that workers are disabled.

**Comment by grahamperrin** [comment]
> I like the new organization 

+1

However, #494

**Comment by grahamperrin** [comment]
https://davidwalsh.name/demo/notifications-api.php

> how it's possible 

Maybe simplest to think of the **Show a Notification** button as you _pulling_ a notice from the server, and you can not click the button after closing the window. 

Service workers are more commonly associated with _push_ …

**Comment by Peacock365** [comment]
@hawkeye116477 Regarding the "New Tab" settings... When you transfer them to Waterfox Current, please note that the search bar can already be disabled in _about:preferences#home_... The "Enable new tab page", assuming that the new tab page flips to _about:blank_ if the setting gets unticked, is also already present in _about:preferences#home_. Therefore, only the "Disable thumbnail capturing on new tab page" setting would be worthwhile to port to Waterfox Current.

The WebRTC / Geolocation / Referrer Header settings would be great to have in Waterfox Current.

**Comment by hawkeye116477** [comment]
@Peacock365 I know, I saw, I already done that on local, not pushed yet only. I'm also working on adding status bar similar as on Vivaldi :smile:.

Update:
```
Maybe simplest to think of the Show a Notification button as you pulling a notice from the server, and you can not click the button after closing the window.

Service workers are more commonly associated with push …
```
Ok, then I'll set `dom.webnotifications.serviceworker.enabled` to false and change that checkbox to set 3 prefs => 2 for notifications and 1 for service worker.

**Comment by grahamperrin** [comment]
i18n: will everything that's proposed work OOTB, or will parts require translation?

**Comment by hawkeye116477** [comment]
@grahamperrin Yes, if you won't translate then it will be in English, only Polish is translated.
I created experimental project on crowdin => https://crowdin.com/project/testwater (expires in 4 days).
Here is repository with unpacked files => https://github.com/hawkeye116477/LanguagePacksForWaterfox.

**Comment by Peacock365** [comment]
But why aren't the new settings translated? This picture shows the picture / Javascript setting translated into Russian, in Cyberfox:

https://www.white-windows.ru/wp-content/uploads/2016/03/4509708_6.png

Also, if the settings are not translated, what is this commit all about?

https://github.com/MrAlex94/Waterfox/pull/1203/commits/b7323f4b3300a12159654435f2a0a99dd9a37ca3

Sorry if the question is silly, I don't get the meaning of the "not translated" statement yet considering the links.

**Comment by hawkeye116477** [comment]
@Peacock365` Also, if the settings are not translated, what is this commit all about?`
I just added English strings to every langpack (expect Polish) and translated only to Polish.

`This picture shows the picture / Javascript setting translated into Russian, in Cyberfox`
Yeah, but I need to view files for all langpacks, also some strings has different entity and for some I changed original string a little. Maybe I'll look to that later, but for now priority is just adding new things.

**Comment by grahamperrin** [comment]
**Many** things to be thankful for in this PR. For now (without testing the effects) I'm just watching with interest. 

For now, a couple of points: 

https://github.com/MrAlex94/Waterfox/pull/1203#issuecomment-547819410

> Old settings are default, 

:+1: 

https://github.com/MrAlex94/Waterfox/pull/1203#issuecomment-548707832

> The … Geolocation … settings would be great to have in Waterfox Current.

Do you mean, the settings that must be modified (on some platforms) for location to be determined? 

I mean, the inability of Waterfox to detect location is sometimes a source of confusion; it will be good to have something without dragons at `about:preferences ` 

> in Waterfox Current.

– in Waterfox Classic, yes?

**Comment by grahamperrin** [comment]
[browser.restart.purgecache – clear fast restart cache on browser restart – Waterfox Classic](https://www.reddit.com/r/waterfox/comments/e3qc2h/browserrestartpurgecache_clear_fast_restart_cache/)

> … a few questions about fast load data. …

**Pull request merged**